garage_0_8: 0.8.5 -> 0.8.6, garage: 0.9.1 -> 0.9.2

RaitoBezarius opened the pull request:
## Description of changes

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.9.2
https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.8.6

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RaitoBezarius committed:
garage_0_8: 0.8.5 -> 0.8.6

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.8.6
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
garage: 0.9.1 -> 0.9.2

https://git.deuxfleurs.fr/Deuxfleurs/garage/releases/tag/v0.9.2
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

CobaltCause reviewed (approved):
I tested 0.9.2 and it works, plus now I can get rid of [this hacky patch](https://or.computer.surgery/charles/servy-fleet/-/blob/0f8fadafaa3e1aafa87c0d1dee15dae7a2aac712/config/garage/cli-rpc-secret-file.patch) which is great.

RaitoBezarius commented:
Result of `nixpkgs-review pr 292619` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_2)</li>
    <li>garage_0_8 (garage_0_8_6)</li>
  </ul>
</details>

adamcstephens reviewed (approved):
Result of `nixpkgs-review pr 292619` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_2)</li>
    <li>garage_0_8 (garage_0_8_6)</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 292619` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_2)</li>
    <li>garage_0_8 (garage_0_8_6)</li>
  </ul>
</details>

RaitoBezarius commented:
Result of `nixpkgs-review pr 292619` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>garage_0_8</li>
    <li>garage_0_8_6</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_2)</li>
  </ul>
</details>

RaitoBezarius commented:
Result of `nixpkgs-review pr 292619` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>garage_0_8</li>
    <li>garage_0_8_6</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>garage (garage_0_9 ,garage_0_9_2)</li>
  </ul>
</details>

RaitoBezarius committed:
garage: mark broken on Darwin

It broke again because of some linker error.
```
garage-aarch64-darwin> warning: use of deprecated associated function `garage_db::sled_adapter::SledDb::init`: The Sled database is now deprecated and will be removed in Garage v1.0. Please migrate to LMDB or Sqlite as soon as possible.
garage-aarch64-darwin>    --> src/model/garage.rs:134:31
garage-aarch64-darwin>     |
garage-aarch64-darwin> 134 |                 db::sled_adapter::SledDb::init(db)
garage-aarch64-darwin>     |                                           ^^^^
garage-aarch64-darwin>     |
garage-aarch64-darwin>     = note: `#[warn(deprecated)]` on by default
garage-aarch64-darwin>
garage-aarch64-darwin>    Compiling garage_api v0.9.2 (/private/tmp/nix-build-garage-0.9.2.drv-0/source/src/api)
garage-aarch64-darwin>    Compiling garage_web v0.9.2 (/private/tmp/nix-build-garage-0.9.2.drv-0/source/src/web)
garage-aarch64-darwin> warning: `garage_model` (lib) generated 1 warning
garage-aarch64-darwin>    Compiling garage v0.9.2 (/private/tmp/nix-build-garage-0.9.2.drv-0/source/src/garage)
garage-aarch64-darwin> warning: use of deprecated associated function `garage_db::sled_adapter::SledDb::init`: The Sled database is now deprecated and will be removed in Garage v1.0. Please migrate to LMDB or Sqlite as soon as possible.
garage-aarch64-darwin>   --> src/garage/cli/convert_db.rs:65:29
garage-aarch64-darwin>    |
garage-aarch64-darwin> 65 |             Ok(sled_adapter::SledDb::init(db))
garage-aarch64-darwin>    |                                      ^^^^
garage-aarch64-darwin>    |
garage-aarch64-darwin>    = note: `#[warn(deprecated)]` on by default
garage-aarch64-darwin>
garage-aarch64-darwin> error: linker `/nix/store/6kpjydf9x7zqa1xq2qipnwr32ki3fs2n-clang-wrapper-16.0.6/bin/cc` not found
garage-aarch64-darwin>   |
garage-aarch64-darwin>   = note: No such file or directory (os error 2)
```

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

adamcstephens reviewed (approved):
Upgraded my three node cluster successfully.

garage cargo:0.9.2 [features: k2v, sled, lmdb, sqlite, consul-discovery, kubernetes-discovery, metrics, telemetry-otlp, bundled-libs]
